### PR TITLE
Add Circle processor

### PR DESCRIPTION
**Pull request by codebrain**
Relates to: https://github.com/elastic/elasticsearch/pull/43851

### CODE REVIEW

File: src/Nest/Ingest/Processors/DissectProcessor.cs [hunk 1]
```diff
@@ -73,6 +73,7 @@ public class DissectProcessorDescriptor<T>
 		public DissectProcessorDescriptor<T> Pattern(string pattern) =>
 			Assign(pattern, (a, v) => a.Pattern = v);
 
+		// TODO! rename the parameter in 8.0
```
Comment (russcam) [inline src/Nest/Ingest/Processors/DissectProcessor.cs]: I think we can rename this here. I understand that it is technically a binary breaking change

Comment (Mpdreamz) [reply src/Nest/Ingest/Processors/DissectProcessor.cs]: Let's rename in 8.0 instead, eventhough impact is small.

**Comment by Mpdreamz** [comment]
1 test failure, unrelated: https://github.com/elastic/elasticsearch-net/issues/4167

File: src/Nest/Ingest/Processors/DissectProcessor.cs (hunk 1 above)
Comment (russcam) [reply src/Nest/Ingest/Processors/DissectProcessor.cs]: ok

**Comment by russcam** [comment]
ported to master

**Pull request merged**